code/ui/manager/src/components/sidebar/TreeNode.tsx

```diff
@@ -145,6 +142,26 @@ export const RootNode = styled.div(({ theme }) => ({
   color: theme.textMutedColor,
 }));
 
+const Wrapper = styled.div({
+  display: 'flex',
+  alignItems: 'center',
+});
+
+const InvisibleText = styled.p({
+  margin: 0,
+  width: 0,
+});
+
+// Make the content have a min-height equal to one line of text
+export const IconsWrapper: FunctionComponent<{ children?: React.ReactNode }> = ({ children }) => {
+  return (
+    <Wrapper>
+      <InvisibleText>&nbsp;</InvisibleText>
```

Comment:
Why do you need this InvisibleText? It should automatically center it in the middle without this line, no?

Reply:
The reason is that when the text for the TreeNode breaks in multiple lines the icon needs to be centered with the first line of text on the right side not with the whole container, like the first example here:
![image](https://github.com/storybookjs/storybook/assets/58397934/a7e4b800-8ef1-47b6-8236-a52d98f68346)
To achieve this the only way I found is to wrap the Icon within an element that has a height of just one line of text and align the icon vertically inside it, so it will be aligned with the first line of text of the TreeNode. We can't use a fixed value that represents the height of one line of text because each font can have a different height, to deal with it I used this trick of placing an invisible character to set a height for the container of the icon to be exactly the same as one line of a text.
Makes sense?
Also, we could align the icon in the center of the TreeNode not with the first line of text, that will be way easier and we can remove the InvisibleText
---
Just for visualization
Aligning with the first line of text:
![Screenshot_10](https://github.com/storybookjs/storybook/assets/58397934/98d7d930-5db2-4c87-8ea5-4b1a6a70fd33)

Aligning with the TreeNode:
![Screenshot_11](https://github.com/storybookjs/storybook/assets/58397934/16560629-e0e1-4646-b3ed-b1f8de6f433d)